typescript/cli/src/validator/preFlightCheck.ts

```diff
@@ -0,0 +1,136 @@
+import { MerkleTreeHook__factory } from '@hyperlane-xyz/core';
+import { HyperlaneCore, S3Validator } from '@hyperlane-xyz/sdk';
+import { Address, eqAddress } from '@hyperlane-xyz/utils';
+
+import { CommandContext } from '../context/types.js';
+import { errorRed, logBlue, logGreen, warnYellow } from '../logger.js';
+
+export const checkValidatorSetup = async (
+  context: CommandContext,
+  chain: string,
+  validators: Set<Address>,
+) => {
+  const { multiProvider, registry } = context;
+
+  const addresses = await registry.getAddresses();
+
+  const core = HyperlaneCore.fromAddressesMap(addresses, multiProvider);
+
+  const validatorAnnounce = core.getContracts(chain).validatorAnnounce;
+  const merkleTreeHook = MerkleTreeHook__factory.connect(
+    addresses[chain].merkleTreeHook,
+    multiProvider.getProvider(chain),
+  );
+
+  let merkleTreeLatestCheckpointIndex: number | undefined;
+  try {
+    const [_, latestCheckpointIndex] = await merkleTreeHook[
+      'latestCheckpoint()'
+    ]();
+
+    merkleTreeLatestCheckpointIndex = latestCheckpointIndex;
+    logBlue(
+      `\nLatest check point index of incremental merkle tree: ${merkleTreeLatestCheckpointIndex}\n`,
+    );
+  } catch (err) {
+    warnYellow(
+      `❗️ Failed to fetch latest checkpoint index of merkleTreeHook on ${chain} \n`,
+    );
+  }
+
+  let announcedValidators;
+  try {
+    announcedValidators = await validatorAnnounce.getAnnouncedValidators();
+  } catch (err) {
+    errorRed(
+      `❌ Failed to fetch announced validators for ${chain}. Exiting.\n`,
+    );
+    process.exit(1);
+  }
+
+  const confirmedValidators = new Set<Address>();
+  const unconfirmedValidators = new Set<Address>();
+
+  for (const validator of validators) {
+    const matches = announcedValidators.filter((address) =>
+      eqAddress(address, validator),
+    );
+    if (matches.length === 0) {
+      unconfirmedValidators.add(validator);
+    } else {
+      confirmedValidators.add(validator);
+    }
+  }
+
+  const errorSet = new Set<string>();
+  errorSet.add('Validator pre flight check failed:');
+
+  if (unconfirmedValidators.size > 0) {
+    errorSet.add('Some validators have not been announced.');
+  }
+
+  let failedToReadError = false;
+  for (const validator of confirmedValidators) {
+    let storageLocations;
+    try {
+      storageLocations = await validatorAnnounce.getAnnouncedStorageLocations([
```

Comment:
nit: instead of n requests and the `getAnnouncedValidators` above, you can get all the info in a single RPC. This is what the relayer does when it's fetching announcements when relaying. Might be useful to avoid rate limiting
```
const validatorStorageLocations = await validatorAnnounce.getAnnouncedStorageLocations(validators);
// Now iterate through validatorStorageLocations, where indices correspond to
// validators, and any validators that haven't been announced will have an empty array
// as their listed storage locations
```

Example output with cast:
```
$ cast call 0xCe74905e51497b4adD3639366708b821dcBcff96 'getAnnouncedStorageLocations(address[])(string[][])' --rpc-url https://ethereum.publicnode.com '[0x94438a7de38d4548ae54df5c6010c4ebc5239eae,0x0000000000000000000000000000000000000000]'
[["s3://dsrv-hyperlane-v3-validator-signatures-validator1-ethereum/eu-central-1"], []]
```

Reply:
I did see this method but was unsure as to whether the response would guarantee ordering. Will make the change